### PR TITLE
change strcpy to strlcpy

### PR DESCRIPTION
**Pull request by flyingfish89**
## Summary
change strcpy->strlcpy

### CODE REVIEW

**Review by acassis — APPROVED** [review]

File: fs/rpmsgfs/rpmsgfs_server.c [hunk 1]
```diff
@@ -607,9 +608,12 @@ static int rpmsgfs_readdir_handler(FAR struct rpmsg_endpoint *ept,
       entry = readdir(dir);
       if (entry)
         {
+          size = MIN(rpmsg_virtio_get_buffer_size(ept->rdev),
+                     rpmsg_virtio_get_rx_buffer_size(ept->rdev));
+          size = MIN(size - len, strlen(entry->d_name) + 1);
```
Comment (xiaoxiang781216) [inline fs/rpmsgfs/rpmsgfs_server.c:613]:
```suggestion
          size = MIN(size - len);
```

File: fs/rpmsgfs/rpmsgfs_server.c [hunk 2]
```diff
@@ -599,6 +599,7 @@ static int rpmsgfs_readdir_handler(FAR struct rpmsg_endpoint *ept,
   FAR struct rpmsgfs_readdir_s *msg = data;
   FAR struct dirent *entry;
   int ret = -ENOENT;
+  size_t size;
```
Comment (xiaoxiang781216) [inline fs/rpmsgfs/rpmsgfs_server.c:602]: move after line 603

File: fs/rpmsgfs/rpmsgfs_client.c [hunk 3]
```diff
@@ -928,8 +921,7 @@ int rpmsgfs_client_chstat(FAR void *handle, FAR const char *path,
   uint32_t space;
   size_t len;
 
-  len  = sizeof(*msg);
-  len += strlen(path) + 1;
+  len  = sizeof(*msg) + strlen(path) + 1;
```
Comment (xiaoxiang781216) [inline fs/rpmsgfs/rpmsgfs_client.c:924]: remove extra space

File: fs/rpmsgfs/rpmsgfs_client.c [hunk 4]
```diff
@@ -889,8 +883,7 @@ int rpmsgfs_client_stat(FAR void *handle, FAR const char *path,
...
```
Comment (xiaoxiang781216) [inline fs/rpmsgfs/rpmsgfs_client.c:886]: remove extra space

File: fs/rpmsgfs/rpmsgfs_client.c [hunk 5]
```diff
@@ -810,8 +806,7 @@ int rpmsgfs_client_mkdir(FAR void *handle, FAR const char *pathname,
   uint32_t space;
   size_t len;
 
-  len  = sizeof(*msg);
-  len += strlen(pathname) + 1;
+  len  = sizeof(*msg) + strlen(pathname) + 1;
```
Comment (xiaoxiang781216) [inline fs/rpmsgfs/rpmsgfs_client.c:809]: remove extra space

File: fs/rpmsgfs/rpmsgfs_client.c [hunk 6]
```diff
@@ -785,8 +782,7 @@ int rpmsgfs_client_unlink(FAR void *handle, FAR const char *pathname)
...
```
Comment (xiaoxiang781216) [inline fs/rpmsgfs/rpmsgfs_client.c:785]: remove extra space

File: fs/rpmsgfs/rpmsgfs_client.c [hunk 7]
```diff
@@ -761,8 +759,7 @@ int rpmsgfs_client_statfs(FAR void *handle, FAR const char *path,
   uint32_t space;
   size_t len;
 
-  len  = sizeof(*msg);
-  len += strlen(path) + 1;
+  len  = sizeof(*msg) + strlen(path) + 1;
```
Comment (xiaoxiang781216) [inline fs/rpmsgfs/rpmsgfs_client.c:762]: remove extra space

File: fs/rpmsgfs/rpmsgfs_client.c [hunk 8]
```diff
@@ -405,8 +405,7 @@ int rpmsgfs_client_open(FAR void *handle, FAR const char *pathname,
   uint32_t space;
   size_t len;
 
-  len  = sizeof(*msg);
-  len += strlen(pathname) + 1;
+  len  = sizeof(*msg) + strlen(pathname) + 1;
```
Comment (xiaoxiang781216) [inline fs/rpmsgfs/rpmsgfs_client.c:408]: remove extra space

**Comment by GUIDINGLI** [comment]
base on https://github.com/apache/nuttx/pull/8629

**Comment by xiaoxiang781216** [comment]
please rebase to the last master.

**Review by GUIDINGLI — APPROVED** [review]

**Pull request merged**